### PR TITLE
BUG: Categorical.fillna with non-category tuple

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/arrays/categorical/test_missing.py [hunk 1]
```diff
@@ -100,6 +100,13 @@ def test_fillna_iterable_category(self, named):
 
         tm.assert_categorical_equal(result, expected)
 
+        # Case where the Point is not among our categories; we want ValueError,
+        #  not NotImplementedError GH#41914
+        cat = Categorical(np.array([Point(1, 0), Point(0, 1), None], dtype=object))
+        msg = "Cannot setitem on a Categorical with a new category"
+        with pytest.raises(ValueError, match=msg):
```
Comment (simonjayhawkins) [inline pandas/tests/arrays/categorical/test_missing.py:107]: This is ValueError whereas in #41919 changed to TypeError with the same message?

Comment (jbrockmendel) [reply pandas/tests/arrays/categorical/test_missing.py:107]: yes, this should eventually be TypeError

File: pandas/tests/arrays/categorical/test_missing.py [hunk 2]
```diff
@@ -100,6 +100,13 @@ def test_fillna_iterable_category(self, named):
 
         tm.assert_categorical_equal(result, expected)
 
+        # Case where the Point is not among our categories; we want ValueError,
+        #  not NotImplementedError GH#41914
+        cat = Categorical(np.array([Point(1, 0), Point(0, 1), None], dtype=object))
+        msg = "Cannot setitem on a Categorical with a new category"
```
Comment (jreback) [inline pandas/tests/arrays/categorical/test_missing.py:106]: is it possible to customize the error messages to the caller method (eg fill a here) as we have a generic validate (good)

can be a follow on

**Review by jreback — APPROVED** [review]

File: pandas/tests/arrays/categorical/test_missing.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/arrays/categorical/test_missing.py:106]: yah thatd be nice.  we did something like that for the datetimelikes

**Pull request merged**